src/Console/Ldap/SynchronizeUsersCommand.php

```diff
@@ -507,4 +566,69 @@ private function validateInput(InputInterface $input)
             }
         }
     }
+
+    /**
+     * Convert the old "single integer" format for the --deleted-user-strategy
+     * option into the new "3 comma-separated integers" format
+     *
+     * @deprecated
+     *
+     * @param int $deleted_user_strategy old format
+     *
+     * @return string new format
+     */
+    protected function convertOldDeletedUserStrategyToNew(int $deleted_user_strategy): string
+    {
+        Toolbox::deprecated("Usage of a deprecated format for the '--deleted-user-strategy' option.");
```

Comment:
Why should this method be deprecated? Old method is deprecated, OK. But this new one used to migrate should not be.
You can of course keep warnings displayed to users, and I guess you can set it private also.

Reply:
We thought with @orthagh that it was a good way to be able to identify the method and make sure we don't forget to drop it in the 10.2 version (or whatever name the N+2 major version will end us having).
It will also log the deprecated error, which can be useful as the output of the command wont be seen if its ran from a crontab.

We did agree at the time that the concept of a new method that is already deprecated was kind of absurd but it seemed to be the best way we could find to handle this.